packages/shared-ux/router/impl/router.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+import {
+  Route as ReactRouterRoute,
+  RouteComponentProps,
+  RouteProps,
+  useRouteMatch,
+} from 'react-router-dom';
+import { useKibanaSharedUX } from './services';
+import { useSharedUXExecutionContext } from './use_execution_context';
+
+/**
+ * This is a wrapper around the react-router-dom Route component that inserts
+ * MatchPropagator in every application route. It helps track all route changes
+ * and send them to the execution context, later used to enrich APM
+ * 'route-change' transactions.
+ */
+export const Route = ({ children, component: Component, render, ...rest }: RouteProps) => {
+  const component = useMemo(() => {
+    if (!Component) {
+      return undefined;
+    }
+    return (props: RouteComponentProps) => (
+      <>
+        <MatchPropagator />
+        <Component {...props} />
+      </>
+    );
+  }, [Component]);
+
+  if (component) {
+    return <ReactRouterRoute {...rest} component={component} />;
+  }
+  if (render) {
+    return (
+      <ReactRouterRoute
+        {...rest}
+        render={(props) => (
+          <>
+            <MatchPropagator />
+            {render(props)}
+          </>
+        )}
+      />
+    );
+  }
+  if (typeof children === 'function') {
```

Comment:
Can we put this and previous "if" in one if statement, cause they seem to be doing exactly the same thing?

Reply:
Wait I could use some clarification - the first if is checking for a render and then rendering props whereas the if on line 54 is checking for the typeof children to be a function and if it is then is using the function from children to render the props (vs render itself). Let me know what I'm missing thanks!

Reply:
Thank you for the help! - 0556d16